hwmv2: atmel_sam0: Port to HWMv2

nandojve opened the pull request:
This ports all the Atmel SAM0 SoC's family and all related boards to HWMv2.

nandojve committed:
boards: samr21_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
soc: atmel: sam0: Port to HWMv2

Port all Atmel SAM0 SoCs to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: samd21_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: samd20_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: saml21_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: samc21n_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: samr34_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: same54_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: seeeduino_xiao: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: serpente: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_feather_m0_basic_proto: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_feather_m0_lora: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_grand_central_m4_express: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_itsybitsy_m4_express: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: adafruit_trinket_m0: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: ev11l78a: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: wio_terminal: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: arduino_zero: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: arduino_mkrzero: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: arduino_nano_33_iot: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

gmarull reviewed (dismissed):
update docs/maintainers as needed

nandojve commented:
> update docs/maintainers as needed

I updated on #68838.

nandojve committed:
samples: atmel_sam0: Update platform name

This update all atmel_sam0 related entries to use the new
platform name.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
tests: atmel_sam0: Update platform name

This update all atmel_sam0 related entries to use the new
platform name.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
tests: atmel_sam: adc: Fix sam4e_xpro adc build

This add a missing sam4e_xpro overlay in the tests/drivers/adc for the
sam4e_xpro board.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve commented:
Rebased and added [0ea3c80](https://github.com/zephyrproject-rtos/zephyr/pull/68841/commits/0ea3c8027d578f5506746b6b5b039a833b514152) to fix the `test/drivers/adc` for the `sam4e_xpro`.

I did not understood how it passed previously without this overlay.
